compiler-plugin/src/main/scala/scalapb/compiler/DescriptorImplicits.scala

```diff
@@ -784,9 +786,18 @@ class DescriptorImplicits(params: GeneratorParams, files: Seq[FileDescriptor]) {
 
     def scalaName: String =
       if (scalaOptions.hasScalaName) scalaOptions.getScalaName
-      else if (enumValue.getFile.scalaOptions.getEnumValueNaming == EnumValueNaming.CAMEL_CASE)
-        allCapsToCamelCase(enumValue.getName, true)
-      else enumValue.getName
+      else {
+        val enumValueName: String =
+          if (enumValue.getFile.scalaOptions.getEnumStripPrefix) {
+            val enumName = enumValue.getType.getName
+            val prefixes = Seq(enumName, NameUtils.toAllCaps(enumName)).flatMap(n => Seq(n + "_", n))
```

Comment:
I prefer to have a simpler rule. Let's keep oen prefix - `NameUtils.toAllCaps(enumName) + "_"` , and omit the other three options. The concern is unintended consequences when there's no intention to strip: `enum C { CAT = 1; }` is not necessarily expected to be stripped to just `AT`.